lib/StackAnalysis/SegregateDirectStackAccesses.cpp

```diff
@@ -0,0 +1,152 @@
+/// \file SegregateDirectStackAccesses.cpp
+/// \brief Segregate direct stack accesses from all other memory accesses
+/// through alias information. This provides a way to say that stack accesses do
+/// not interfere with any other memory access.
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/MDBuilder.h"
+#include "llvm/IR/PatternMatch.h"
+
+#include "revng/StackAnalysis/SegregateDirectStackAccesses.h"
+#include "revng/Support/Assert.h"
+
+using namespace llvm;
+using SDSAP = SegregateDirectStackAccessesPass;
+
+PreservedAnalyses SDSAP::run(Function &F, FunctionAnalysisManager &FAM) {
+  Context = &(F.getContext());
+
+  // Get the result of the GCBI analysis
+  auto &M = *F.getParent();
+  const auto &MAMProxy = FAM.getResult<ModuleAnalysisManagerFunctionProxy>(F);
+  GCBI = MAMProxy.getCachedResult<GeneratedCodeBasicInfoAnalysis>(M);
+  if (!GCBI)
+    GCBI = &(FAM.getResult<GeneratedCodeBasicInfoAnalysis>(F));
```

Comment:
Is this pass actually used both from a module-pass manager and function-pass manager?

Reply:
It is.

Reply:
Nope, only from a FunctionPass Manager.

Reply:
Then drop the logic for handling both.